pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -302,7 +302,7 @@ public SuccessResponse updateTableConfig(
 
       ensureMinReplicas(tableConfig);
       verifyTableConfigs(tableConfig);
-      _pinotHelixResourceManager.setExistingTableConfig(tableConfig, tableNameWithType, tableType);
+      _pinotHelixResourceManager.updateTableConfig(tableConfig, tableNameWithType, tableType);
```

Comment:
We've already have a method called `verifyTableConfigs` in Line 304 and there's a new method `validateTableTenantConfig ` in `updateTableConfig`. Can they be merged together?

Reply:
verifyTableCOnfigs compares the offline and realtime table configs for hybrid tables.
validateTableTenantCOnfig is specifically for validating tenant configs.
I intentionally pushed the validation inside the update method, because update and add are called from multiple files (v1 and v2 configs). And it was better to have the call to validate in 1 place instead of in every endpoint.
Maybe the ensureMinReplicas and verify can also be pushed down. But that would need more evaluation, because all tests use the add and update and setExisting methods, which do not want all the validations